__tests__/getApiTopics.test.js

```diff
@@ -0,0 +1,39 @@
+const app = require("../app");
+const request = require("supertest");
+const data = require("../db/data/test-data/index");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+
+beforeEach(() => seed(data));
+afterAll(() => db.end());
+
+describe("GET /api/topics", () => {
+  test("should respond with a 200 status", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+  test("should respond with an array of topic", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toHaveProperty("topics", expect.any(Array));
+      });
+  });
+  test("Array should contain object containing correct properties", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(body).toHaveProperty("topics", expect.any(Array));
+        expect(typeof topics).toBe("object");
+        topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug", expect.any(String));
+          expect(topic).toHaveProperty("description", expect.any(String));
+        });
+      });
+  });
+  test("should respond with a 404 status if invalid endpoint", () => {
+    return request(app).get("/api/topic").expect(404);
+  });
+});
```

Comment:
As this test is generic, could we pop this in a separate describe block for errors please? 👍